File: development/NET/samples/server/ServerDLL/VB/OpcProject/MyDaAddressSpaceRoot.vb
Imports Softing.OPCToolbox
Imports Softing.OPCToolbox.Server

Namespace DllSample

	Public Class MyDaAddressSpaceRoot
		Inherits DaAddressSpaceRoot

		Public Overloads Overrides Function QueryAddressSpaceElementData(ByVal elementId As String, ByRef anAddressSpaceElement As AddressSpaceElement) As Integer
			'  TODO: add string based address space validations
			anAddressSpaceElement = Nothing
			Return CType(EnumResultCode.E_NOTIMPL, Integer)
		End Function

		Public Overrides Function QueryAddressSpaceElementChildren(ByVal anElementID As String, ByVal aChildrenList As System.Collections.ArrayList) As Integer

		End Function
	End Class
End Namespace

File: development/C++/samples/server/SoftingServer/src/XXOPC/MyDaAddressSpaceRoot.cpp
#include "stdafx.h"
#include "MyDaAddressSpaceRoot.h"
#include "OpcServer.h"
#include "IOManager.h"


using namespace SoftingOPCToolboxServer;

//-----------------------------------------------------------------------------
// MyDaAddressSpaceRoot
//-----------------------------------------------------------------------------

long MyDaAddressSpaceRoot::queryAddressSpaceElementData(
	tstring& anElementID,
	AddressSpaceElement* anElement)
{
	//  TODO: add string based address space validations
	return E_NOTIMPL;
}   //  end QueryAddressSpaceElementData

void MyDaAddressSpaceRoot::activationStateUpdated(
	std::vector<DaAddressSpaceElement*>& anElementList)
{
	getOpcServer()->getIOManager()->cyclicStateChanged(anElementList);
}
